cpp/arcticdb/processing/operation_dispatch_ternary.cpp

```diff
@@ -0,0 +1,582 @@
+                    condition,
+                    offset_string.offset(),
+                    EmptyResult{},
+                    *output_column);
+        } else if constexpr (is_numeric_type(val_type_info::data_type) || is_bool_type(val_type_info::data_type)) {
+            using TargetType = val_type_info::RawType;
+            output_column = std::make_unique<Column>(val.type(), Sparsity::PERMITTED);
+            auto value = static_cast<TargetType>(val.get<typename val_type_info::RawType>());
+            value_string = fmt::format("{}", value);
+            ternary_transform<typename val_type_info::TDT, arguments_reversed>(
+                    condition,
+                    value,
+                    EmptyResult{},
+                    *output_column);
+        } else {
+            user_input::raise<ErrorCode::E_INVALID_USER_ARGUMENT>("Invalid ternary operator arguments {}",
+                                                                  ternary_operation_with_types_to_string<arguments_reversed>(
+                                                                          val.to_string<typename val_type_info::RawType>(),
+                                                                          val.type(),
+                                                                          "",
+                                                                          {}));
+        }
+    });
+    return {ColumnWithStrings(std::move(output_column), string_pool, ternary_operation_column_name<arguments_reversed>(value_string, ""))};
+}
+
+template VariantData ternary_operator<true>(const util::BitSet& condition, const Value& val, EmptyResult);
+template VariantData ternary_operator<false>(const util::BitSet& condition, const Value& val, EmptyResult);
+
+// This handles the filter case where we select based on two fixed bool values. This could be produced from Python like:
+// lazy_df = lazy_df[where(lazy_df["col1"] < 0, True, False)]
+// although this is dumb, as the same effect could be achieved with:
+// lazy_df = lazy_df[lazy_df["col1"] < ]
+// It is actually here to cope with cases where input processing produced FullResult or EmptyResult, which are then
+// combined with a fixed bool value
+VariantData ternary_operator(const util::BitSet& condition, bool left, bool right) {
+    util::BitSet output_bitset;
+    if (left && right) {
+        if (condition.size() > 0) {
+            output_bitset.set_range(0, condition.size() - 1);
+        }
+    } else if (left) {
+        // right is false
+        output_bitset = condition;
+    } else if (right) {
+        // left is false
+        output_bitset = ~condition;
+    } // else left and right are false, the resize below will correctly initialise all bits to zero
+    output_bitset.resize(condition.size());
+    return output_bitset;
+}
+
+VariantData visit_ternary_operator(const VariantData& condition, const VariantData& left, const VariantData& right) {
```

Comment:
This function has lost me - I'll pick through it some more and we can chat about it tomorrow